sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/BatchCommitLog.scala

```diff
@@ -45,28 +45,37 @@ import org.apache.spark.sql.SparkSession
 class BatchCommitLog(sparkSession: SparkSession, path: String)
   extends HDFSMetadataLog[String](sparkSession, path) {
 
+  import BatchCommitLog._
+
+  def add(batchId: Long): Unit = {
+    super.add(batchId, SERIALIZED_VOID)
+  }
+
+  override def add(batchId: Long, metadata: String): Boolean = {
+    throw new UnsupportedOperationException(
+      "BatchCommitLog does not take any metadata, use 'add(batchId)' instead")
+  }
+
   override protected def deserialize(in: InputStream): String = {
     // called inside a try-finally where the underlying stream is closed in the caller
     val lines = IOSource.fromInputStream(in, UTF_8.name()).getLines()
     if (!lines.hasNext) {
       throw new IllegalStateException("Incomplete log file in the offset commit log")
     }
-    parseVersion(lines.next().trim, BatchCommitLog.VERSION)
+    parseVersion(lines.next.trim, VERSION)
     // read metadata
-    lines.next().trim match {
-      case BatchCommitLog.SERIALIZED_VOID => null
-      case metadata => metadata
-    }
+    val metadata = lines.next.trim
+    assert(metadata == SERIALIZED_VOID, s"Batch commit log has unexpected metadata: $metadata ")
```

Comment:
As long as the version is the same, we should be able to ignore this. If we don't want to ignore it, then in the message we should say please upgrade your Spark version or something